src/core/pattern/decision_book.cc

```diff
@@ -38,21 +39,33 @@ Decision DecisionBookField::nextDecision(const CoreField& cf, const KumipuyoSeq&
     if (!matcher.match(pattern_, cf))
         return Decision();
 
+    // Now it is guaranteed that field is matched.
+
+    // Check sequence with 1 Tsumo.
     const Kumipuyo& kp1 = seq.get(0);
-    const Kumipuyo& kp2 = seq.get(1);
+    for (const auto& entry : decisions1_) {
+        if (matchNext(&matcher, entry.first, kp1))
+            return entry.second;
+        if (kp1.isRep())
+            continue;
+        if (matchNext(&matcher, entry.first, kp1.reverse()))
+            return entry.second.reverse();
+    }
 
-    // Field matched. check next sequence.
-    for (const auto& entry : decisions_) {
+    // Check sequence with 2 Tsumos.
```

Comment:
Maybe good to check 2-tsumo version first?

So, when we specify
```
AA = [1, 2]
AAAB = [2, 3]
```
`AAAB` should be `[2, 3]` and `AA**` should be `[1, 2]` ?